### PR TITLE
:sparkles: clientgen: add a cluster-delegating fake client

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
needs https://github.com/kcp-dev/client-go/pull/6

**Comment by ncdc** [comment]
/approve

Will lgtm after the dependent PR merges & we update go.mod

**Review by ncdc — APPROVED** [review]

**Pull request merged**
